clients/src/main/java/org/apache/kafka/common/requests/ProduceRequest.java

```diff
@@ -69,11 +74,6 @@ public ProduceRequest build(short version) {
             return build(version, true);
         }
 
-        // Visible for testing only
-        public ProduceRequest buildUnsafe(short version) {
```

Comment:
This is unused.

Reply:
It seems we can merge `private ProduceRequest build(short version, boolean validate)` into `public ProduceRequest build(short version)`. Otherwise, IDE complains the method has a "always true" input variable `validate`.

Reply:
Will include this in my next batch of changes.